packages/checkbox-group/src/vaadin-checkbox-group.js

```diff
@@ -253,6 +254,11 @@ class CheckboxGroup extends FieldMixin(FocusMixin(DisabledMixin(ElementMixin(The
     } else if (this.value.includes(checkbox.value)) {
       checkbox.checked = true;
     }
+
+    const tooltip = this._tooltipController.node;
+    if (tooltip) {
+      setAriaIDReference(checkbox.inputElement, 'aria-describedby', { newId: tooltip._uniqueId });
+    }
```

Comment:
This relies on timing where the tooltip is already available when the checkboxes are added. I'm not sure if we can rely on that in Flow's case (and in general).
For example, if you build the group programmatically like this, the linking doesn't work as expected:

```js
const checkboxGroup = document.createElement('vaadin-checkbox-group');
checkboxGroup.label = 'Preferred language of contact:';
document.body.appendChild(checkboxGroup);

const checkbox1 = document.createElement('vaadin-checkbox');
checkbox1.value = 'en';
checkbox1.label = 'English';
checkboxGroup.appendChild(checkbox1);

const checkbox2 = document.createElement('vaadin-checkbox');
checkbox2.value = 'fr';
checkbox2.label = 'Français';
checkboxGroup.appendChild(checkbox2);

const tooltip = document.createElement('vaadin-tooltip');
tooltip.slot = 'tooltip';
tooltip.text = 'If not selected, English is used';
checkboxGroup.appendChild(tooltip);
```

Reply:
Would it not work if CBG listened to slot changes in the `tooltip` slot?

Reply:
Yes, that's an option too, but `TooltipController` is based on `SlotController` which already does that.

Reply:
Updated the PR to use `tooltip-changed` event and set `ariaTarget` on the lazily added tooltip.
Also, added new IT file `checkbox-group-tooltip.test.js` using the above code snippet.